### PR TITLE
fix metrics query bug for pods with duplicate name in one node

### PR DESCRIPTION
**Pull request by junotx**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

fix metrics query bug for pods with duplicate name in one node

/area monitoring
/cc @benjaminhuo

### CODE REVIEW

File: pkg/simple/client/monitoring/prometheus/promql.go [hunk 1]
```diff
@@ -365,17 +360,56 @@ func makePodMetricExpr(tmpl string, o monitoring.QueryOptions) string {
...
+						strings.Join(namespaces, "|"), resourcesFilter, o.NodeName)
+				}
+			}
 		} else {
-			podSelector = fmt.Sprintf(`pod=~"%s", node="%s"`, o.ResourceFilter, o.NodeName)
+			// For monitoring pods in the whole cluster
+			// Get /pods
+			resourcesFilter := o.ResourceFilter
+			if len(pods) > 0 {
+				resourcesFilter = fmt.Sprintf("(%s)|(%s)", resourcesFilter, strings.Join(pods, "|"))
```
Comment (benjaminhuo) [inline pkg/simple/client/monitoring/prometheus/promql.go:402]: /pods API is newly added, do we still need to add the content of old resourcesFilter to the content of namespacedResourceFilter ?

It's ok to do so for /nodes/{node}/pods API for backward-compatible.

**Review by benjaminhuo — CHANGES_REQUESTED** [review]

File: pkg/kapis/monitoring/v1alpha3/register.go [hunk 2]
```diff
@@ -313,7 +316,8 @@ func AddToContainer(c *restful.Container, k8sClient kubernetes.Interface, monito
 
 	ws.Route(ws.GET("/nodes/{node}/pods/{pod}").
 		To(h.handlePodMetricsQuery).
-		Doc("Get pod-level metric data of a specific pod. Navigate to the pod by the node where it is scheduled.").
+		Deprecate().
+		Doc("Get pod-level metric data of a specific pod. Navigate to the pod by the node where it is scheduled. deprecated: please use `/namespaces/{namespace}/pods/{pod}` instead.").
```
[the remark this answers is not in the document]
Comment (junotx) [reply pkg/kapis/monitoring/v1alpha3/register.go:320]: There is no `/pods/{pod}`

[the remark this answers is not in the document]
Comment (benjaminhuo) [reply pkg/kapis/monitoring/v1alpha3/register.go]: I mean the new added  /pods API should be able to get metrics for a specific pod, so the desc could be changed to 

Deprecated: please use /namespaces/{namespace}/pods/{pod} or /pods ?

**Review by benjaminhuo — APPROVED** [review]
/lgtm

**Review by benjaminhuo — APPROVED** [review]

**Comment by zryfish** [comment]
/unapprove

@junotx  conflicts

**Review by benjaminhuo — APPROVED** [review]

**Pull request merged**